apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -1504,34 +1418,8 @@ func (p *ProvisionerAPIV7) ModificationStatus(_, _ struct{}) {}
 
 // ModificationStatus returns the modification status for each given entity.
 // Only machine tags are accepted.
-func (p *ProvisionerAPI) ModificationStatus(args params.Entities) (params.StatusResults, error) {
-	result := params.StatusResults{
-		Results: make([]params.StatusResult, len(args.Entities)),
-	}
-	canAccess, err := p.getAuthFunc()
-	if err != nil {
-		logger.Errorf("failed to get an authorisation function: %v", err)
-		return result, errors.Trace(err)
-	}
-	for i, arg := range args.Entities {
-		mTag, err := names.ParseMachineTag(arg.Tag)
-		if err != nil {
-			logger.Warningf("ModificationStatus called with %q which is not a valid machine tag: %v", arg.Tag, err)
-			result.Results[i].Error = common.ServerError(common.ErrPerm)
-			continue
-		}
-		machine, err := p.getMachine(canAccess, mTag)
-		if err == nil {
-			var statusInfo status.StatusInfo
-			statusInfo, err = machine.ModificationStatus()
-			result.Results[i].Status = statusInfo.Status.String()
-			result.Results[i].Info = statusInfo.Message
-			result.Results[i].Data = statusInfo.Data
-			result.Results[i].Since = statusInfo.Since
-		}
-		result.Results[i].Error = common.ServerError(err)
-	}
-	return result, nil
+func (p *ProvisionerAPIV8) ModificationStatus(args params.Entities) (params.StatusResults, error) {
```

Comment:
So older provisioner clients don't call this anymore? Not being fully across the work to date, I'm not sure why we need to keep v8 WatchModelMachinesCharmProfiles() etc but not this v8 method as well.

Reply:
I chatted with Heather about this. We have explicitly chosen for the Profile work not *not* be backwards compatible. And instead, we have code that checks you can't upgrade a charm to a version that has a Profile *if* you are running in mixed mode. (This was done, because the old code was watching for documents that we just aren't creating now.)

We don't want those agents to bounce when they call in, we just don't have any work for them to do (which we've ensured with the 'upgrade-charm' code.)

I think this is ok. We'd *love* to return NotSupported from the other ones to make it clear that the particular version of the API isn't available anymore, but that causes the workers to bounce. This doesn't have a valid return, and as long as the Watcher code doesn't give them a reason to look at this (since it is disabled), there shouldn't be a problem.


We should document the API compatibility break in a comment.

Reply:
It appears this is dead code and was never called in 2.5.2, 2.5.3, 2.5.4, nor tip of 2.5.  It was added in when SetModificationStatus() was added in 2.5.2.

Reply:
If this is dead code, can we not remove it?